server.py

```diff
@@ -68,68 +69,44 @@ def read_file():
-                concurrent_connections = metadata.get_concurrent_connections(filename)
-                closest_servers = dict()
-                for concurrent_connection in concurrent_connections:
-                    closest_server = util.find_closest_servers_with_ip(concurrent_connection)
-                    if closest_server not in closest_servers:
-                        closest_servers[closest_server] = 1
-                    else:
-                        closest_servers[closest_server] += 1
-                target_server = max(closes_servers)
-                # 2) Check if there is enough space on the remote server.
-                url = 'http://%s/can_move_file?%s' % (target_server, urllib.urlencode({ 'uuid': filename }))
-                response = requests.get(url)
-                if response.status_code == 200:
-                    # 3) Copy the file to that server.
-                    clone_file(request.args.get('uuid'), target_server, 'DISTRIBUTED_REPLICATE', ip_address)
-        else:
-            raise Exception('Something fishy is going on... Should have at least one request')
-
-        # remove the number of concurrent requests to the file
-        @after_this_request
-        def remove_request(response):
-            metadata.remove_concurrent_request(filename, ip_address)
-            metadata.close()
 
     file_path = UPLOAD_FOLDER + '/' + secure_filename(filename)
     if (metadata.is_file_exist_locally(filename, app.config['HOST']) is not None):
-        host_address = app.config['simulation_ip'] if 'simulation_ip' in app.config else app.config('HOST')
-        logger.log(filename, ip_address, host_address, 'READ', 200, os.path.getsize(file_path))
+        if app.config['use_dist_replication']:
+            distributed_replication(filename, ip_address, delay_time, metadata)
+            # remove the number of concurrent requests to the file
+            @after_this_request
+            def remove_request(response):
+                metadata.remove_concurrent_request(filename, ip_address)
+                metadata.close()
+
+        host_address = app.config['simulation_ip'] if 'simulation_ip' in app.config else app.config['HOST']
+        logger.log(filename, ip_address, host_address, 'READ', requests.code.ok, os.path.getsize(file_path))
         time.sleep(delay_time)
         return send_from_directory(UPLOAD_FOLDER, secure_filename(filename))
 
     redirect_address = metadata.lookup_file(filename, app.config['HOST'])
     if (redirect_address is not None):
-        host_address = app.config['simulation_ip'] if 'simulation_ip' in app.config else app.config('HOST')
-        url = 'http://%s/read?%s' % (redirect_address[0], urllib.urlencode({ 'uuid': filename, 'ip': host_address }))
-        logger.log(filename, ip_address, host_address, 'READ', 302, -1)
-        return redirect(url, code=302)
+        host_address = app.config['simulation_ip'] if 'simulation_ip' in app.config else app.config['HOST']
```

Comment:
I don't know if this is difficult, but I feel like this and the ~20 lines below it can be written a little differently to prevent redundant code.

Instead of having multiple return/log statements, could it be something more like:
* Check metadata
* If metadata is empty, start checking other servers
* If found in other server, update metadata
* Log and redirect

Just a suggestion! :santa: